### PR TITLE
Changing collation to utf8mb4

### PR DESCRIPTION
**Pull request by yvesh**
Pull Request for Issue #150

#### Summary of Changes

Updates the default collation for new installations to utf8mb4. Added an update script (version 3.5.1) for previous installations.

#### Testing Instructions

**Installing:** 
Just build the Web Links package (or use the attached one) and install normally.

**Updating:** 
Build the package with version 3.5.1 in `jorobo.ini` (or use the attached one) and update with the Joomla! installer. Check if the collation of #__weblinks and the alias column has changed. 

screenshot 2016-06-26 14 32 20

[pkg-weblinks-3.5.1.zip](https://github.com/joomla-extensions/weblinks/files/333736/pkg-weblinks-3.5.1.zip)

### CODE REVIEW

**Comment by zero-24** [comment]
@yvesh what is about the mysql versions (client & server) that can't handle uft8mb4?

see: https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/script.php#L1684-L1805

**Comment by zero-24** [comment]
as well as: https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/database/driver/mysqli.php#L954

**Comment by yvesh** [comment]
@zero-24 you are right. Already suppressed that mess from my memories.. :-) Going to update the PR to check that in the script.php before..

File: src/administrator/components/com_weblinks/script.php [hunk 1]
```diff
@@ -239,4 +242,62 @@ private function addColumnsIfNeeded()
 			$db->execute();
 		}
 	}
+
+	/**
+	 * Method to convert #__weblinks to utf8mb4 on mysql (if possible).
+	 *
```
Comment (zero-24) [inline src/administrator/components/com_weblinks/script.php]: The doc block is missing for $parent ;)

**Comment by mbabker** [comment]
You don't need to do this as a complex conditional in your PHP update script.  You can make it a regular SQL schema update and since the MySQL drivers downgrade gracefully from utf8mb4 to utf8 if the former's support isn't there you don't need to do any conditionals around things.

See https://github.com/BabDev/Podcast-Manager/commit/39e8904f8779cd6bcb05debd99c49b2fc37ce476 for how I did it in an extension.

**Comment by yvesh** [comment]
@mbabker That's what i did before until @zero-24 complained.. Then i revert it back.. :)

**Comment by mbabker** [comment]
The core does it in a massively complex way for reasons I never fully grasped or paid attention to.  At an extension level, if you have to do it with PHP scripts then you're doing something majorly wrong.

**Comment by zero-24** [comment]
Ok sorry I don't know that behaviors in the MySQL servers.

**Comment by yvesh** [comment]
Okay reverted back to initial version

**Comment by chrisdavenport** [comment]
I have tested this item :white_check_mark: successfully on c6821b079fc2782ad6f68a66c2423b64805b2337<br /><br />New install and upgrade both work correctly.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/226">issues.joomla.org/weblinks/226</a>.</sub>

**Comment by pritalpatel** [comment]
Thanks, I install the package its work correctly.
weblink

**Comment by pritalpatel** [comment]
I have tested this item :white_check_mark: successfully on c6821b079fc2782ad6f68a66c2423b64805b2337<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/226">issues.joomla.org/weblinks/226</a>.</sub>

**Comment by infograf768** [comment]
What will happen if a < 3.6.0 site updates weblinks with a change to utf8mb4 ?

**Comment by mbabker** [comment]
Technically this would only be valid against a Joomla 3.5 or later install.  So an install script and the update server definition should limit presenting this update to only those versions (i.e. Joomla 3.4 shouldn't be able to update to it).

**Comment by yvesh** [comment]
Okay i am going to change it a third time (had that check already in script.php..). 

@mbabker @chrisdavenport @infograf768 Any plans for how long Weblinks should support < 3.5.0 installs?

**Comment by chrisdavenport** [comment]
So all we need to do is bump the version requirement in the XML files to 3.5 then?

**Comment by javigomez** [comment]
Please @yvesh update your pull with latest `master` to see if Travis passes this time.

Thanks

**Comment by yvesh** [comment]
@javigomez Done, passing now :)

**Comment by chrisdavenport** [comment]
I have tested this item :white_check_mark: successfully on 5cbb43d76c4571cef87e9f215be7e44f03260b79<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/226">issues.joomla.org/weblinks/226</a>.</sub>

**Comment by Ruchiranga** [comment]
I have tested this item :white_check_mark: successfully on 5cbb43d76c4571cef87e9f215be7e44f03260b79<br /><br />Success<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/226">issues.joomla.org/weblinks/226</a>.</sub>

**Pull request merged**